gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoRestSerializer.java

```diff
@@ -159,4 +219,23 @@ public String toBody(String fieldName, RequestT fieldValue) {
   public String toBody(String fieldName, RequestT fieldValue, boolean numericEnum) {
     return toJson(fieldValue, numericEnum);
   }
+
+  /**
+   * Serializes an object to a query parameter Handles the case of a message such as Duration,
+   * FieldMask or Int32Value to prevent wrong formatting that String.valueOf() would make
+   *
+   * @param fieldValue a field value to serialize
+   */
+  public String toQueryParamValue(Object fieldValue) {
+    // This will match with message types that are serializable (e.g. FieldMask)
+    if (fieldValue instanceof GeneratedMessageV3 && !isNonSerializableMessageValue(fieldValue)) {
```

Comment:
I would just use `Message` instead of `GeneratedMessagev3` in all places since it's more generic. I also found that  we could pass an Enum to query params, where Enum is not of `Message` type, so we may need to handle enum specifically as well.